keps/sig-cluster-lifecycle/kubeadm/20190424-kubeadm-for-windows.md

```diff
@@ -115,23 +115,35 @@ The motivation for this KEP is to provide a tool to allow users to take a Window
 
 A user will have a Windows machine that they want to join to an existing Kubernetes cluster.
 
-1. The user will download a set of required binaries (kubelet, kube-proxy, kubeadm, kubectl, Flannel) using a script. The same script will also wrap kubeadm execution.
+1. The user will download a set of required binaries (kubelet, kubeadm, and wins) using a script.
 
-2. The script will register kubelet and kube-proxy as Windows services.
+1. The script will register kubelet as a Windows service.
 
-3. The script will upload a default ConfigMap to the cluster for the default Windows KubeletConfiguration if one does not already exist.
+1. The user will run "kubeadm join ..." to join the node to the cluster. In this step kubeadm will run preflight checks and proceed with the regular join process.
 
-4. The script will run "kubeadm join ..." to join the node to the cluster. In this step kubeadm will run preflight checks and proceed with the regular join process.
+1. kubeadm will restart the kubelet service using flags that kubeadm fed to the Windows service. kubeadm will proceed to bootstrap the node. After this process is finished the node should show with the status of NotReady.
 
-5. kubeadm will restart the kubelet service using flags that kubeadm fed to the Windows service. kubeadm will proceed to bootstrap the node. After this process is finished the node should show with the status of NotReady.
+1. The user will then deploy the Flannel and kube-proxy DaemonSets. These will run and initialize container networking.
```

Comment:
i guess these have to be downloaded separately?
possibly need to reside in the same folder as the script and the script should download them?

proposing this structure:
```
kubeadm
  v1.18
    - script
    - kube-proxy.yaml
    - flannel.yaml
  v1.19
    - script
    - kube-proxy.yaml
    - flannel.yaml
...
```

i'm not proposing sub folders for the patch releases for now, due to the fact that the [sig-windows-tools repo](https://github.com/kubernetes-sigs/sig-windows-tools) is not very actively maintained. 

also something that we didn't discuss during the kubedam office hours of 15th Jan is do we want to have the new way of deploying in a separate folder.

given this is alpha we can do the following:
move `sig-windows-tools/kubeadm/` to `sig-windows-tools/kubeadm-legacy`

start applying the new logic in `sig-windows-tools/kubeadm/`.

Reply:
I mention elsewhere that I don't think we should need to rev the script for each Kubernetes version. I am hoping that we can have a parameter for the desired version, and then the rest of it ideally doesn't have any changes in between versions. Maybe we can start with that hope for a single script, and if we do later discover some backwards incompatible changes we need to make to it that we figure something out then?

We do need to version the kube-proxy DS though.

Reply:
i think we should version it right away, even if we just have to copy it, because we don't know when the kubelet flags might get deprecated. it's better to have this early IMO.